packages/aws_common/lib/src/io/aws_file_platform_html.dart

```diff
@@ -81,10 +82,18 @@ class AWSFilePlatform extends AWSFile {
   final Stream<List<int>>? _stream;
   int? _size;
   Blob? _resolvedBlobFromPath;
+  // this field may be assigned when resolve a blob from path
   String? _contentType;
 
   @override
-  String? get contentType => _contentType ?? super.contentType;
+  String? get contentType {
+    final file = _inputFile ?? _inputBlob;
+    if (file != null) {
+      return file.type;
+    }
+
+    return _contentType ?? super.contentType;
```

Comment:
`package:mime` works on web - why not use that here?

Reply:
In Web, the path is a url that can resolve a blob from, `lookupMimeType` returns `null` in this case, as its implementation is extracting the extension name from the path string to determine the mime type...